### DISCUSSION TITLE
Help needed in understanding how to set up C++ on my Mac

### DISCUSSION THREAD

**Discussion by 112Butter** [microsoft/vscode-cpptools#10586] [Q&A]
I've been trying to start learning how to program on C++.
I tried all kinds of instructions on how to set it up using Visual Code, yet I don't seem to get anywhere, and I feel like I'm about to lose it after working on it for 8 hours. :(

Currently I'm following the instruction on https://code.visualstudio.com/docs/cpp/config-clang-mac.
After getting to the step [Run helloworld.cpp], 4. I seem to get stuck.
I'm getting the error message "expected ';' at end of declaration ggc[row 9, column 23]"

My tasks.json file does also look completely different:
{
    "tasks": [
        {
            "type": "cppbuild",
            "label": "C/C++: clang++ Aktive Datei kompilieren",
            "command": "/usr/bin/clang++",
            "args": [
                "-fcolor-diagnostics",
                "-fansi-escape-codes",
                "-g",
                "${file}",
                "-o",
                "${fileDirname}/${fileBasenameNoExtension}"
            ],
            "options": {
                "cwd": "${fileDirname}"
            },
            "problemMatcher": [
                "$gcc"
            ],
            "group": {
                "kind": "build",
                "isDefault": true
            },
            "detail": "Vom Debugger generierte Aufgabe."
        }
    ],
    "version": "2.0.0"
}

I want to set it up to be able to follow the tutorial https://www.youtube.com/watch?v=8jLOx1hD3_o&t=5508s, which is supposed to be a good course in understanding the basics, but I feel like I need 31 hours to just set C++ up. 

I hope I gave all the information that is needed to help me with my issue. If not, I definitely don't mind answering any questions.
Any help would be extremely appreciated!!

**Comment by sean-mcmanus** [comment]
Can you add `"-std=c++17",` to your args list? See https://code.visualstudio.com/docs/cpp/config-clang-mac#_troubleshooting
